Building fails, if path contains spaces

Building in "F:\A B\ponyc\" with make.bat -v

I'm guessing the arguments passed to the linker are missing quotes.
Because it seems like it tries to treat the folder as a lib file.

Results in:
```
PS F:\A B\ponyc> ./make -v
Waf: Entering directory `F:\A B\ponyc\build\release-llvm-3.9.1'
[168/227] Linking build\release-llvm-3.9.1\ponyc.exe
17:43:57 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\amd64\\LINK.exe', '/nologo', '/MANIFEST', '/NXCOMPAT', '/SUBSYSTEM:CONSOLE', 'src\\ponyc\\main.c.8.o', '/OUT:F:\\A B\\ponyc\\build\\release-llvm-3.9.1\\ponyc.exe', '/LIBPATH:.', 'libponyc.lib', 'libponyrt.lib', '/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\LIB\\amd64', '/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\ATLMFC\\LIB\\amd64', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\lib\\10.0.14393.0\\ucrt\\x64', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\NETFXSDK\\4.6.1\\lib\\um\\x64', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\lib\\10.0.14393.0\\um\\x64', '/LIBPATH:C:\\Windows\\Microsoft.NET\\Framework64\\v4.0.30319', '/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\LIB\\amd64', '/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\ATLMFC\\LIB\\amd64', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\UnionMetadata', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\References', '/LIBPATH:C:\\Program Files (x86)\\Microsoft SDKs\\Windows Kits\\10\\ExtensionSDKs\\Microsoft.VCLibs\\14.0\\References\\CommonConfiguration\\neutral', '/LIBPATH:[]', '/LIBPATH:F:\\A B\\ponyc\\build\\release-llvm-3.9.1\\ThirdParty\\lib\\LLVM-3.9.1-Release\\lib', 'F:\\A.lib', 'LLVMPasses.lib', 'F:\\A.lib', 'LLVMInterpreter.lib', 'F:\\A.lib', 'LLVMCoverage.lib', 'F:\\A.lib', 'LLVMLTO.lib', 'F:\\A.lib', 'LLVMObjCARCOpts.lib', 'F:\\A.lib', 'LLVMOrcJIT.lib', 'F:\\A.lib', 'LLVMLibDriver.lib', 'F:\\A.lib', 'LLVMOption.lib', 'F:\\A.lib', 'LLVMXCoreDisassembler.lib', 'F:\\A.lib', 'LLVMXCoreCodeGen.lib', 'F:\\A.lib', 'LLVMXCoreDesc.lib', 'F:\\A.lib', 'LLVMXCoreInfo.lib', 'F:\\A.lib', 'LLVMXCoreAsmPrinter.lib', 'F:\\A.lib', 'LLVMSystemZDisassembler.lib', 'F:\\A.lib', 'LLVMSystemZCodeGen.lib', 'F:\\A.lib', 'LLVMSystemZAsmParser.lib', 'F:\\A.lib', 'LLVMSystemZDesc.lib', 'F:\\A.lib', 'LLVMSystemZInfo.lib', 'F:\\A.lib', 'LLVMSystemZAsmPrinter.lib', 'F:\\A.lib', 'LLVMSparcDisassembler.lib', 'F:\\A.lib', 'LLVMSparcCodeGen.lib', 'F:\\A.lib', 'LLVMSparcAsmParser.lib', 'F:\\A.lib', 'LLVMSparcDesc.lib', 'F:\\A.lib', 'LLVMSparcInfo.lib', 'F:\\A.lib', 'LLVMSparcAsmPrinter.lib', 'F:\\A.lib', 'LLVMPowerPCDisassembler.lib', 'F:\\A.lib', 'LLVMPowerPCCodeGen.lib', 'F:\\A.lib', 'LLVMPowerPCAsmParser.lib', 'F:\\A.lib', 'LLVMPowerPCDesc.lib', 'F:\\A.lib', 'LLVMPowerPCInfo.lib', 'F:\\A.lib', 'LLVMPowerPCAsmPrinter.lib', 'F:\\A.lib', 'LLVMNVPTXCodeGen.lib', 'F:\\A.lib', 'LLVMNVPTXDesc.lib', 'F:\\A.lib', 'LLVMNVPTXInfo.lib', 'F:\\A.lib', 'LLVMNVPTXAsmPrinter.lib', 'F:\\A.lib', 'LLVMMSP430CodeGen.lib', 'F:\\A.lib', 'LLVMMSP430Desc.lib', 'F:\\A.lib', 'LLVMMSP430Info.lib', 'F:\\A.lib', 'LLVMMSP430AsmPrinter.lib', 'F:\\A.lib', 'LLVMMipsDisassembler.lib', 'F:\\A.lib', 'LLVMMipsCodeGen.lib', 'F:\\A.lib', 'LLVMMipsAsmParser.lib', 'F:\\A.lib', 'LLVMMipsDesc.lib', 'F:\\A.lib', 'LLVMMipsInfo.lib', 'F:\\A.lib', 'LLVMMipsAsmPrinter.lib', 'F:\\A.lib', 'LLVMHexagonDisassembler.lib', 'F:\\A.lib', 'LLVMHexagonCodeGen.lib', 'F:\\A.lib', 'LLVMHexagonAsmParser.lib', 'F:\\A.lib', 'LLVMHexagonDesc.lib', 'F:\\A.lib', 'LLVMHexagonInfo.lib', 'F:\\A.lib', 'LLVMBPFCodeGen.lib', 'F:\\A.lib', 'LLVMBPFDesc.lib', 'F:\\A.lib', 'LLVMBPFInfo.lib', 'F:\\A.lib', 'LLVMBPFAsmPrinter.lib', 'F:\\A.lib', 'LLVMARMDisassembler.lib', 'F:\\A.lib', 'LLVMARMCodeGen.lib', 'F:\\A.lib', 'LLVMARMAsmParser.lib', 'F:\\A.lib', 'LLVMARMDesc.lib', 'F:\\A.lib', 'LLVMARMInfo.lib', 'F:\\A.lib', 'LLVMARMAsmPrinter.lib', 'F:\\A.lib', 'LLVMAMDGPUDisassembler.lib', 'F:\\A.lib', 'LLVMAMDGPUCodeGen.lib', 'F:\\A.lib', 'LLVMipo.lib', 'F:\\A.lib', 'LLVMVectorize.lib', 'F:\\A.lib', 'LLVMLinker.lib', 'F:\\A.lib', 'LLVMIRReader.lib', 'F:\\A.lib', 'LLVMAMDGPUAsmParser.lib', 'F:\\A.lib', 'LLVMAMDGPUDesc.lib', 'F:\\A.lib', 'LLVMAMDGPUInfo.lib', 'F:\\A.lib', 'LLVMAMDGPUAsmPrinter.lib', 'F:\\A.lib', 'LLVMAMDGPUUtils.lib', 'F:\\A.lib', 'LLVMAArch64Disassembler.lib', 'F:\\A.lib', 'LLVMAArch64CodeGen.lib', 'F:\\A.lib', 'LLVMGlobalISel.lib', 'F:\\A.lib', 'LLVMAArch64AsmParser.lib', 'F:\\A.lib', 'LLVMAArch64Desc.lib', 'F:\\A.lib', 'LLVMAArch64Info.lib', 'F:\\A.lib', 'LLVMAArch64AsmPrinter.lib', 'F:\\A.lib', 'LLVMAArch64Utils.lib', 'F:\\A.lib', 'LLVMX86Disassembler.lib', 'F:\\A.lib', 'LLVMX86AsmParser.lib', 'F:\\A.lib', 'LLVMMCJIT.lib', 'F:\\A.lib', 'LLVMExecutionEngine.lib', 'F:\\A.lib', 'LLVMRuntimeDyld.lib', 'F:\\A.lib', 'LLVMMIRParser.lib', 'F:\\A.lib', 'LLVMAsmParser.lib', 'F:\\A.lib', 'LLVMX86CodeGen.lib', 'F:\\A.lib', 'LLVMX86Desc.lib', 'F:\\A.lib', 'LLVMX86Info.lib', 'F:\\A.lib', 'LLVMMCDisassembler.lib', 'F:\\A.lib', 'LLVMX86AsmPrinter.lib', 'F:\\A.lib', 'LLVMX86Utils.lib', 'F:\\A.lib', 'LLVMSelectionDAG.lib', 'F:\\A.lib', 'LLVMAsmPrinter.lib', 'F:\\A.lib', 'LLVMCodeGen.lib', 'F:\\A.lib', 'LLVMTarget.lib', 'F:\\A.lib', 'LLVMScalarOpts.lib', 'F:\\A.lib', 'LLVMInstCombine.lib', 'F:\\A.lib', 'LLVMInstrumentation.lib', 'F:\\A.lib', 'LLVMTransformUtils.lib', 'F:\\A.lib', 'LLVMBitWriter.lib', 'F:\\A.lib', 'LLVMAnalysis.lib', 'F:\\A.lib', 'LLVMProfileData.lib', 'F:\\A.lib', 'LLVMSymbolize.lib', 'F:\\A.lib', 'LLVMDebugInfoPDB.lib', 'F:\\A.lib', 'LLVMDebugInfoCodeView.lib', 'F:\\A.lib', 'LLVMDebugInfoDWARF.lib', 'F:\\A.lib', 'LLVMObject.lib', 'F:\\A.lib', 'LLVMMCParser.lib', 'F:\\A.lib', 'LLVMMC.lib', 'F:\\A.lib', 'LLVMBitReader.lib', 'F:\\A.lib', 'LLVMCore.lib', 'F:\\A.lib', 'LLVMLineEditor.lib', 'F:\\A.lib', 'LLVMObjectYAML.lib', 'F:\\A.lib', 'LLVMTableGen.lib', 'F:\\A.lib', 'LLVMSupport.lib', 'kernel32.lib', 'user32.lib', 'gdi32.lib', 'winspool.lib', 'comdlg32.lib', 'advapi32.lib', 'shell32.lib', 'ole32.lib', 'oleaut32.lib', 'uuid.lib', 'odbc32.lib', 'odbccp32.lib', 'vcruntime.lib', 'ucrt.lib', 'Ws2_32.lib', 'dbghelp.lib', 'Shlwapi.lib']
LINK : fatal error LNK1181: cannot open input file 'F:\A.lib'

* Node F:\A B\ponyc\build\release-llvm-3.9.1\src\libponyrt\lang\except_try_catch.o is created more than once (full message on 'waf -v -v'). The task generators are:
  1. 'libponyrt' in F:\A B\ponyc
  2. 'libponyrt' in F:\A B\ponyc
If you think that this is an error, set no_errcheck_out on the task instance
Waf: Leaving directory `F:\A B\ponyc\build\release-llvm-3.9.1'
Build failed
 -> task in 'ponyc' failed with exit status 1181:
        {task 2121443584936: cxxprogram main.c.8.o -> ponyc.exe,ponyc.exe.manifest}
['C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\amd64\\LINK.exe', '/nologo', '/MANIFEST', '/NXCOMPAT', '/SUBSYSTEM:CONSOLE', 'src\\ponyc\\main.c.8.o', '/OUT:F:\\A B\\ponyc\\build\\release-llvm-3.9.1\\ponyc.exe', '/LIBPATH:.', 'libponyc.lib', 'libponyrt.lib', '/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\LIB\\amd64', '/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\ATLMFC\\LIB\\amd64', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\lib\\10.0.14393.0\\ucrt\\x64', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\NETFXSDK\\4.6.1\\lib\\um\\x64', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\lib\\10.0.14393.0\\um\\x64', '/LIBPATH:C:\\Windows\\Microsoft.NET\\Framework64\\v4.0.30319', '/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\LIB\\amd64', '/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\ATLMFC\\LIB\\amd64', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\UnionMetadata', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\References', '/LIBPATH:C:\\Program Files (x86)\\Microsoft SDKs\\Windows Kits\\10\\ExtensionSDKs\\Microsoft.VCLibs\\14.0\\References\\CommonConfiguration\\neutral', '/LIBPATH:[]', '/LIBPATH:F:\\A B\\ponyc\\build\\release-llvm-3.9.1\\ThirdParty\\lib\\LLVM-3.9.1-Release\\lib', 'F:\\A.lib', 'LLVMPasses.lib', 'F:\\A.lib', 'LLVMInterpreter.lib', 'F:\\A.lib', 'LLVMCoverage.lib', 'F:\\A.lib', 'LLVMLTO.lib', 'F:\\A.lib', 'LLVMObjCARCOpts.lib', 'F:\\A.lib', 'LLVMOrcJIT.lib', 'F:\\A.lib', 'LLVMLibDriver.lib', 'F:\\A.lib', 'LLVMOption.lib', 'F:\\A.lib', 'LLVMXCoreDisassembler.lib', 'F:\\A.lib', 'LLVMXCoreCodeGen.lib', 'F:\\A.lib', 'LLVMXCoreDesc.lib', 'F:\\A.lib', 'LLVMXCoreInfo.lib', 'F:\\A.lib', 'LLVMXCoreAsmPrinter.lib', 'F:\\A.lib', 'LLVMSystemZDisassembler.lib', 'F:\\A.lib', 'LLVMSystemZCodeGen.lib', 'F:\\A.lib', 'LLVMSystemZAsmParser.lib', 'F:\\A.lib', 'LLVMSystemZDesc.lib', 'F:\\A.lib', 'LLVMSystemZInfo.lib', 'F:\\A.lib', 'LLVMSystemZAsmPrinter.lib', 'F:\\A.lib', 'LLVMSparcDisassembler.lib', 'F:\\A.lib', 'LLVMSparcCodeGen.lib', 'F:\\A.lib', 'LLVMSparcAsmParser.lib', 'F:\\A.lib', 'LLVMSparcDesc.lib', 'F:\\A.lib', 'LLVMSparcInfo.lib', 'F:\\A.lib', 'LLVMSparcAsmPrinter.lib', 'F:\\A.lib', 'LLVMPowerPCDisassembler.lib', 'F:\\A.lib', 'LLVMPowerPCCodeGen.lib', 'F:\\A.lib', 'LLVMPowerPCAsmParser.lib', 'F:\\A.lib', 'LLVMPowerPCDesc.lib', 'F:\\A.lib', 'LLVMPowerPCInfo.lib', 'F:\\A.lib', 'LLVMPowerPCAsmPrinter.lib', 'F:\\A.lib', 'LLVMNVPTXCodeGen.lib', 'F:\\A.lib', 'LLVMNVPTXDesc.lib', 'F:\\A.lib', 'LLVMNVPTXInfo.lib', 'F:\\A.lib', 'LLVMNVPTXAsmPrinter.lib', 'F:\\A.lib', 'LLVMMSP430CodeGen.lib', 'F:\\A.lib', 'LLVMMSP430Desc.lib', 'F:\\A.lib', 'LLVMMSP430Info.lib', 'F:\\A.lib', 'LLVMMSP430AsmPrinter.lib', 'F:\\A.lib', 'LLVMMipsDisassembler.lib', 'F:\\A.lib', 'LLVMMipsCodeGen.lib', 'F:\\A.lib', 'LLVMMipsAsmParser.lib', 'F:\\A.lib', 'LLVMMipsDesc.lib', 'F:\\A.lib', 'LLVMMipsInfo.lib', 'F:\\A.lib', 'LLVMMipsAsmPrinter.lib', 'F:\\A.lib', 'LLVMHexagonDisassembler.lib', 'F:\\A.lib', 'LLVMHexagonCodeGen.lib', 'F:\\A.lib', 'LLVMHexagonAsmParser.lib', 'F:\\A.lib', 'LLVMHexagonDesc.lib', 'F:\\A.lib', 'LLVMHexagonInfo.lib', 'F:\\A.lib', 'LLVMBPFCodeGen.lib', 'F:\\A.lib', 'LLVMBPFDesc.lib', 'F:\\A.lib', 'LLVMBPFInfo.lib', 'F:\\A.lib', 'LLVMBPFAsmPrinter.lib', 'F:\\A.lib', 'LLVMARMDisassembler.lib', 'F:\\A.lib', 'LLVMARMCodeGen.lib', 'F:\\A.lib', 'LLVMARMAsmParser.lib', 'F:\\A.lib', 'LLVMARMDesc.lib', 'F:\\A.lib', 'LLVMARMInfo.lib', 'F:\\A.lib', 'LLVMARMAsmPrinter.lib', 'F:\\A.lib', 'LLVMAMDGPUDisassembler.lib', 'F:\\A.lib', 'LLVMAMDGPUCodeGen.lib', 'F:\\A.lib', 'LLVMipo.lib', 'F:\\A.lib', 'LLVMVectorize.lib', 'F:\\A.lib', 'LLVMLinker.lib', 'F:\\A.lib', 'LLVMIRReader.lib', 'F:\\A.lib', 'LLVMAMDGPUAsmParser.lib', 'F:\\A.lib', 'LLVMAMDGPUDesc.lib', 'F:\\A.lib', 'LLVMAMDGPUInfo.lib', 'F:\\A.lib', 'LLVMAMDGPUAsmPrinter.lib', 'F:\\A.lib', 'LLVMAMDGPUUtils.lib', 'F:\\A.lib', 'LLVMAArch64Disassembler.lib', 'F:\\A.lib', 'LLVMAArch64CodeGen.lib', 'F:\\A.lib', 'LLVMGlobalISel.lib', 'F:\\A.lib', 'LLVMAArch64AsmParser.lib', 'F:\\A.lib', 'LLVMAArch64Desc.lib', 'F:\\A.lib', 'LLVMAArch64Info.lib', 'F:\\A.lib', 'LLVMAArch64AsmPrinter.lib', 'F:\\A.lib', 'LLVMAArch64Utils.lib', 'F:\\A.lib', 'LLVMX86Disassembler.lib', 'F:\\A.lib', 'LLVMX86AsmParser.lib', 'F:\\A.lib', 'LLVMMCJIT.lib', 'F:\\A.lib', 'LLVMExecutionEngine.lib', 'F:\\A.lib', 'LLVMRuntimeDyld.lib', 'F:\\A.lib', 'LLVMMIRParser.lib', 'F:\\A.lib', 'LLVMAsmParser.lib', 'F:\\A.lib', 'LLVMX86CodeGen.lib', 'F:\\A.lib', 'LLVMX86Desc.lib', 'F:\\A.lib', 'LLVMX86Info.lib', 'F:\\A.lib', 'LLVMMCDisassembler.lib', 'F:\\A.lib', 'LLVMX86AsmPrinter.lib', 'F:\\A.lib', 'LLVMX86Utils.lib', 'F:\\A.lib', 'LLVMSelectionDAG.lib', 'F:\\A.lib', 'LLVMAsmPrinter.lib', 'F:\\A.lib', 'LLVMCodeGen.lib', 'F:\\A.lib', 'LLVMTarget.lib', 'F:\\A.lib', 'LLVMScalarOpts.lib', 'F:\\A.lib', 'LLVMInstCombine.lib', 'F:\\A.lib', 'LLVMInstrumentation.lib', 'F:\\A.lib', 'LLVMTransformUtils.lib', 'F:\\A.lib', 'LLVMBitWriter.lib', 'F:\\A.lib', 'LLVMAnalysis.lib', 'F:\\A.lib', 'LLVMProfileData.lib', 'F:\\A.lib', 'LLVMSymbolize.lib', 'F:\\A.lib', 'LLVMDebugInfoPDB.lib', 'F:\\A.lib', 'LLVMDebugInfoCodeView.lib', 'F:\\A.lib', 'LLVMDebugInfoDWARF.lib', 'F:\\A.lib', 'LLVMObject.lib', 'F:\\A.lib', 'LLVMMCParser.lib', 'F:\\A.lib', 'LLVMMC.lib', 'F:\\A.lib', 'LLVMBitReader.lib', 'F:\\A.lib', 'LLVMCore.lib', 'F:\\A.lib', 'LLVMLineEditor.lib', 'F:\\A.lib', 'LLVMObjectYAML.lib', 'F:\\A.lib', 'LLVMTableGen.lib', 'F:\\A.lib', 'LLVMSupport.lib', 'kernel32.lib', 'user32.lib', 'gdi32.lib', 'winspool.lib', 'comdlg32.lib', 'advapi32.lib', 'shell32.lib', 'ole32.lib', 'oleaut32.lib', 'uuid.lib', 'odbc32.lib', 'odbccp32.lib', 'vcruntime.lib', 'ucrt.lib', 'Ws2_32.lib', 'dbghelp.lib', 'Shlwapi.lib']
PS F:\A B\ponyc>
```

--- a/wscript
+++ b/wscript
@@ -253,10 +253,7 @@ def build(ctx):
             ctx.env.LLVM_LLC = os.path.join(llvmDir, 'bin', 'llc.exe')
             llvmLibFiles = cmd_output([llvmConfig, '--libs'])
             import re
-            if ctx.options.llvm.startswith(('3.7', '3.8')):
-                llvmLibs = [re.sub(r'-l([^\\\/)]+)', r'\1', x) for x in llvmLibFiles.split(' ')]
-            else:
-                llvmLibs = [re.sub(r'.*[\\\/]([^\\\/)]+)\.lib', r'\1', x) for x in llvmLibFiles.split(' ')]
+            llvmLibs = [re.sub(r'.*[\\\/]([^\\\/)]+)$', r'\1', x) for x in llvmLibFiles.split('.lib') if x]
 
             llvmBuildMode = cmd_output([llvmConfig, '--build-mode'])
             if llvmBuildMode == 'Release':
@@ -398,7 +395,7 @@ def test(ctx):
 
     ctx(
         features = 'seq',
-        rule     = os.path.join(ctx.bldnode.abspath(), 'ponyc') + ' -d -s --checktree --verify ../../packages/stdlib',
+        rule     = '"' + os.path.join(ctx.bldnode.abspath(), 'ponyc') + '" -d -s --checktree --verify ../../packages/stdlib',
         target   = stdlibTarget,
         source   = ctx.bldnode.ant_glob('ponyc*') + ctx.path.ant_glob('packages/**/*.pony'),
     )
@@ -406,7 +403,7 @@ def test(ctx):
     # grammar file
     ctx(
         features = 'seq',
-        rule     = os.path.join(ctx.bldnode.abspath(), 'ponyc') + ' --antlr > pony.g.new',
+        rule     = '"' + os.path.join(ctx.bldnode.abspath(), 'ponyc') + '" --antlr > pony.g.new',
         target   = 'pony.g.new',
     )
 

